server/controllers/install.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @overview Install
+ *
+ * @description
+ * This module is responsible for setting up a new bhima instance
+ * by configuring administrator user, enterprise, project, etc.
+ *
+ * @requires db
+ */
+
+const Q = require('q');
+const db = require('../lib/db');
+const BadRequest = require('../lib/errors/BadRequest');
+
+/**
+ * @method basicInstallExist
+ *
+ * @description
+ * check if the basic information for installation exist, in the case
+ * it doesn't exist we can perform a new installation
+ */
+function basicInstallExist() {
+  // check users, enterprise, projects
+  const userExist = 'SELECT COUNT(*) AS total_users FROM user;';
+  const enterpriseExist = 'SELECT COUNT(*) AS total_enterprises FROM enterprise;';
+  const projectExist = 'SELECT COUNT(*) AS total_projects FROM project;';
```

Comment:
I know JavaScript will treat 1 as `true`, but why don't you just go ahead and make the SQL return a boolean?  Like:
```js
const userExist = 'SELECT COUNT(*) > 1 AS hasUsers FROM user;'
```
or something?